pkg/database/resync.go

```diff
@@ -206,12 +207,79 @@ func upsertResources(resyncBody []byte, clusterName string, syncResponse *model.
 				}
 				incomingUIDs = append(incomingUIDs, uid)
 			}
-			return incomingUIDs, nil
+			err = dao.checkHubClusterRename(ctx, resource, clusterName)
+			return incomingUIDs, err
 		}
 	}
 	return incomingUIDs, nil
 }
 
+func (dao *DAO) checkHubClusterRename(ctx context.Context, resource model.Resource, requestCluster string) error {
+	// check if hub cluster name(s) has changed by comparing existing cluster name
+	//   where _hubClusterResource is true against new resource cluster where _hubClusterResource is true
+	// we check for multiple distinct old hub cluster names in case of the off chance multiple renames have happened
+	//   prior to this old name cleaning
+	if _, ok := resource.Properties["_hubClusterResource"]; ok {
+		// SELECT distinct cluster FROM search.resources WHERE data ? '_hubClusterResource'
+		sql, args, err := goqu.From(goqu.S("search").Table("resources")).
+			Select("cluster").
+			Distinct().
+			Where(goqu.L("???", goqu.C("data"), goqu.Literal("?"), "_hubClusterResource")).ToSQL()
+		if err != nil {
+			klog.Errorf("Error creating query to check existing hub cluster name")
+			return err
+		}
+		rows, err := dao.pool.Query(ctx, sql, args...)
+		if err != nil {
+			klog.Errorf("Error while fetching hub cluster name from database: %s", err.Error())
+			return err
+		}
+
+		clustersToDelete := make([]string, 0)
```

Comment:
Maybe the name `existingHubClusterNames` is more accurate?  This will also have the current hub name.

Reply:
Nevermind. Now I saw the full logic.